sdk/go/pulumi/rpc_test.go

```diff
@@ -880,3 +885,56 @@ func TestDependsOnComponent(t *testing.T) {
 	_, deps = newResource("resI", DependsOn([]Resource{resG}))
 	assert.Equal(t, []string{"resG"}, deps)
 }
+
+func TestOutputValueMarshalling(t *testing.T) {
+	ctx, err := NewContext(context.Background(), RunInfo{})
+	assert.Nil(t, err)
+
+	values := []struct {
+		value    interface{}
+		expected resource.PropertyValue
+	}{
+		{value: nil, expected: resource.NewNullProperty()},
+		{value: 0, expected: resource.NewNumberProperty(0)},
+		{value: 1, expected: resource.NewNumberProperty(1)},
+		{value: "", expected: resource.NewStringProperty("")},
+		{value: "hi", expected: resource.NewStringProperty("hi")},
+		{value: map[string]string{}, expected: resource.NewObjectProperty(resource.PropertyMap{})},
+		{value: []string{}, expected: resource.NewArrayProperty(nil)},
+	}
+	for _, value := range values {
+		for _, deps := range [][]resource.URN{nil, {"fakeURN1", "fakeURN2"}} {
+			for _, known := range []bool{true, false} {
+				for _, secret := range []bool{true, false} {
+					var resources []Resource
+					if len(deps) > 0 {
+						for _, dep := range deps {
+							resources = append(resources, ctx.newDependencyResource(URN(dep)))
+						}
+					}
+
+					out := ctx.newOutput(anyOutputType, resources...)
+					out.getState().resolve(value.value, known, secret, nil)
+
+					expected := resource.Output{
+						Known:        known,
+						Secret:       secret,
+						Dependencies: deps,
+					}
+					if known {
+						expected.Element = value.expected
+					}
+
+					name := fmt.Sprintf("value=%v, known=%v, secret=%v, deps=%v", value, known, secret, deps)
+					t.Run(name, func(t *testing.T) {
+						inputs := Map{"value": out}
+						expected := resource.PropertyMap{"value": resource.NewOutputProperty(expected)}
+						actual, _, _, err := marshalInputs(inputs)
```

Comment:
There's no checks on the `_` and `_` - secret and deps, but perhaps this is okay/trivial.